project/cmake/scripts/common/addon-helpers.cmake

```diff
@@ -40,6 +45,18 @@ macro (build_addon target prefix libs)
     SET_TARGET_PROPERTIES(${target} PROPERTIES PREFIX "lib")
   ENDIF(OS STREQUAL "android")
 
+  SET(LIBRARY_LOCATION $<TARGET_FILE:${target}>)
+  SET(LIBRARY_FILENAME $<TARGET_FILE_NAME:${target}>)
+
+  # if there's an addon.xml.in we need to generate the addon.xml
+  IF(EXISTS ${PROJECT_SOURCE_DIR}/${target}/addon.xml.in)
+    SET(PLATFORM ${CORE_SYSTEM_NAME})
+
+    FILE(READ ${PROJECT_SOURCE_DIR}/${target}/addon.xml.in addon_file)
+    STRING(CONFIGURE ${addon_file} addon_file_conf @ONLY)
```

Comment:
We most likely need to put double quotes around `${addon_file}` and `${addon_file_conf}` to tell CMake that these are strings and not lists (delimited by semicolons).